api/src/main/java/net/md_5/bungee/api/connection/ProxiedPlayer.java

```diff
@@ -290,6 +290,105 @@
      */
     void sendTitle(Title title);
 
+    /**
+     * Sends a resource pack to this player.
+     * <p>
+     * Hash is the SHA-1 digest of the resource pack. It's
+     * recommended to specify this. A hash can be identified by
+     * using the service http://onlinemd5.com/
+     * </p>
+     *
+     * @param url url of resource pack
+     * @param hash optional resource pack hash
+     * @throws IllegalArgumentException if hash is longer than 40
+     * digits
+     */
+    default void sendResourcePack(String url, String hash) throws IllegalArgumentException
+    {
+        sendResourcePack( url, hash, false );
+    }
+
+    /**
+     * Sends a resource pack to this player.
+     * <p>
+     * Hash is the SHA-1 digest of the resource pack. It's
+     * recommended to specify this. A hash can be identified by
+     * using the service http://onlinemd5.com/
+     * </p>
+     *
+     * @param url url of resource pack
+     * @param hash optional resource pack hash
+     * @param forced whether the notchian client will be forced
+     * to use the resource pack. Works 1.17+ only
+     * @throws IllegalArgumentException if hash is longer than 40
+     * digits
+     */
+    default void sendResourcePack(String url, String hash, boolean forced) throws IllegalArgumentException
+    {
+        sendResourcePack( url, hash, forced, null );
+    }
+
+    /**
+     * Sends a resource pack to this player.
+     * <p>
+     * Hash is the SHA-1 digest of the resource pack. It's
+     * recommended to specify this. A hash can be identified by
+     * using the service http://onlinemd5.com/
+     * </p>
+     *
+     * @param url url of resource pack
+     * @param hash optional resource pack hash
+     * @param promptMessage the message shown in the prompt
+     * making the client accept or decline the resource pack.
+     * Works 1.17+ only
+     * @throws IllegalArgumentException if hash is longer than 40
+     * digits
+     */
+    default void sendResourcePack(String url, String hash, BaseComponent[] promptMessage) throws IllegalArgumentException
+    {
+        sendResourcePack( url, hash, false, promptMessage );
+    }
+
+    /**
+     * Sends a resource pack to this player.
+     * <p>
+     * Hash is the SHA-1 digest of the resource pack. It's
+     * recommended to specify this. A hash can be identified by
+     * using the service http://onlinemd5.com/
+     * </p>
+     *
+     * @param url url of resource pack
+     * @param hash optional resource pack hash
+     * @param forced whether the notchian client will be forced
+     * to use the resource pack. Works 1.17+ only
+     * @param promptMessage the message shown in the prompt
+     * making the client accept or decline the resource pack.
+     * Works 1.17+ only
+     * @throws IllegalArgumentException if hash is longer than 40
+     * digits
+     */
+    void sendResourcePack(String url, String hash, boolean forced, BaseComponent[] promptMessage) throws IllegalArgumentException;
```

Comment:
Don't other methods use varargs for this?

Though I note Spigot goes hash, message, forced.

Reply:
![image](https://user-images.githubusercontent.com/37958105/154234464-64da484e-4d63-4036-805d-f8fd6dfefd2c.png)

difference is spigot does a byte array for hash and a string for the prompt